### ISSUE TITLE
BUG: qhull incorrect tests if the point is inside convex hull

### ISSUE THREAD

**Issue by gf0507033** [scipy/scipy#21279]
### Describe your issue.

`scipy.interpolate.LinearNDInterpolator` should be defined on each input point. Apparently for some input points that's not true. The reason is wrong calculation inside qhull if point is inside convex hull of input points or not.

Sample data for reproducing error are [here](https://gist.github.com/gf0507033/2cfba1709e68c6422d8bcb206d957975).

### Reproducing Code Example

```python
import json

from scipy.interpolate import LinearNDInterpolator

import numpy as np

with open('bug.json', 'r') as f:
    points2d = np.array(json.load(f))    

r = np.ones_like(points2d[:, 0])
r_predictor = LinearNDInterpolator(points2d, r)
assert (~np.isnan(r_predictor(points2d))).all()
```

### Error message

```shell
AssertionError                            Traceback (most recent call last)
File /home/grigory/bug.py:12
     10 r = np.ones_like(points2d[:, 0])
     11 r_predictor = LinearNDInterpolator(points2d, r)
---> 12 assert (~np.isnan(r_predictor(points2d))).all()
```

### SciPy/NumPy/Python version and system information

```shell
1.10.1 1.26.4 sys.version_info(major=3, minor=10, micro=14, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: cmake
    found: true
    include directory: unknown
    lib directory: unknown
    name: OpenBLAS
    openblas configuration: unknown
    pc file directory: unknown
    version: 0.3.18
  lapack:
    detection method: cmake
    found: true
    include directory: unknown
    lib directory: unknown
    name: OpenBLAS
    openblas configuration: unknown
    pc file directory: unknown
    version: 0.3.18
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 0.29.33
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.2.1
  pythran:
    include directory: /tmp/pip-build-env-xob7rtip/overlay/lib/python3.10/site-packages/pythran
    version: 0.12.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /opt/python/cp310-cp310/bin/python
  version: '3.10'
```

**Comment by lucascolley** [comment]
Could you be more specific about what you think the problem with qhull is?

**Comment by gf0507033** [comment]
@lucascolley the reason for `np.nan` appearence is [this part](https://github.com/scipy/scipy/blob/main/scipy/interpolate/interpnd.pyx#L344-L354). So the function `qhull._find_simplex` gives incorrect result in this particular case. It's not easy to debug why, it happens deeply inside qhull code.

**Comment by ev-br** [comment]
Indeed there's a single `nan` in a corner of the convex hull (the star shows `outp = interp(points2d); points2d[np.isnan(outp)] `):

![Figure_1](https://github.com/user-attachments/assets/5e63b0b7-1b37-4865-99a6-649fcd5616a2)

changing the interpolator to `interp = LinearNDInterpolator(points2d, values, fill_value=-101)` makes the offending value to be -101 instead of a nan, so it's likely that `qhull._find_simplex` reports that the point is outside. 

The routine is here: https://github.com/scipy/scipy/blob/main/scipy/spatial/_qhull.pyx#L1470. It's in cython, so it's possible to either walk in gdb or insert enough diagnostic prints to pinpoint what goes wrong.

**Comment by gf0507033** [comment]
@ev-br yes, you are right, there is a single nan. I made the same experiment with `fill_value`, so I blamed `qhull._find_simplex`.

You are right, smth bad is happening inside cython code, not inside qhull c code. So I'll try to localize the problem.

**Comment by tylerjereddy** [comment]
It looks to me like there's query between a generator and its membership in the convex hull, which returns a "distance" of `-0.000000000000035390` while `-eps` is `-0.000000000000022204`.  The bailout happens after the eps check here: https://github.com/scipy/scipy/blob/main/scipy/spatial/_qhull.pyx#L1434. When I excise the reproducer into `spatial` alone and use the public `find_simplex`, it suffices to increase the size of `tol`. I'll need to think about this a bit I guess, I suppose the argument is that you should get "exactly" `0.0` for a comparison of the generator with itself on the hull/simplex.

![image](https://github.com/user-attachments/assets/f3985acf-24a8-4c4c-8362-663217f1e8b0)

Is the behavior here actually unexpected?

**Comment by ev-br** [comment]
> Is the behavior here actually unexpected? I guess it is a bit weird to count the hull vertex outside.

Same here. I'd naively expect that the data points are inside not outside. Again very naively, this feels like some sort of numerical error with a weirdly shaped simplex (three data points cluster together at the corner?)

**Comment by DWesl** [comment]
Do you think this is the same problem as #22831?  That example might be easier to step through

**Comment by ev-br** [comment]
With gh-22831, using `interp(points2d, simplex_tolerance=3)` avoids the nan in the corner.
